android/app/src/main/java/com/example/snifferapp/MainActivity.java

```diff
@@ -20,23 +29,43 @@ public void configureFlutterEngine(@NonNull FlutterEngine flutterEngine) {
     new MethodChannel(flutterEngine.getDartExecutor().getBinaryMessenger(), CHANNEL)
             .setMethodCallHandler(
               (call, result) -> {
-                  if (call.method.equals("getArpTable")) {
-                    String arpTableContent = "";
-                    ArrayList<String> arpEntries = new ArrayList<String>();
-                    try {
-                        String line;
-                        BufferedReader br = new BufferedReader(new FileReader("/proc/net/arp"));
-                        while ((line = br.readLine()) != null) {
-                            arpTableContent += line;
-                            arpEntries.add(line);
-                        }
-                        result.success(arpEntries);
-//                        result.success(arpTableContent);
-                    } catch (Exception e) {
-                        result.error("UNAVAILBLE", e.toString(), e);
-                    }
-                  } else {
-                      result.notImplemented();
+                  try {
+                      if (call.method.equals("getArpTable")) {
+                          String arpTableContent = "";
+                          ArrayList<String> arpEntries = new ArrayList<String>();
+
+                          String line;
+                          BufferedReader br = new BufferedReader(new FileReader("/proc/net/arp"));
+                          while ((line = br.readLine()) != null) {
+                              arpTableContent += line;
+                              arpEntries.add(line);
+                          }
+                          result.success(arpEntries);
+
+                      } else if (call.method.equals("getNetworkInterfaceInfo")) {
+                          ConnectivityManager cm = (ConnectivityManager)getContext().getSystemService(Context.CONNECTIVITY_SERVICE);
+                          Network[] networks = cm.getAllNetworks();
+                          ArrayList<String> networkStrings = new ArrayList<String>();
+
+                          for (int i = 0; i < networks.length; i++) {
+                              networkStrings.add("|");
+                              LinkProperties lp = cm.getLinkProperties(networks[i]);
+                              networkStrings.add(lp.getInterfaceName());
+                              // Retrieve network addresses
+                              for (LinkAddress la : lp.getLinkAddresses()) {
+                                  networkStrings.add(la.toString());
+                              }
+                              // Retrieve routing table
+//                              for (RouteInfo ri : lp.getRoutes()) {
```

Comment:
We should probably delete these before merging

Reply:
Right wil do